packages/aws-cdk-lib/core/lib/private/synthesis.ts

```diff
@@ -254,6 +263,97 @@ function invokeAspects(root: IConstruct) {
   }
 }
 
+/**
+ * Invoke aspects V2 runs a stabilization loop and allows Aspects to invoke other Aspects.
+ * Runs if the feature flag '@aws-cdk/core:aspectStabilization' is enabled.
+ *
+ * Unlike the original function, this function does not emit a warning for ignored aspects, since this
+ * function invokes Aspects that are created by other Aspects.
+ *
+ * Throws an error if the function attempts to invoke an Aspect on a node that has a lower priority value
+ * than the most recently invoked Aspect on that node.
+ */
+function invokeAspectsV2(root: IConstruct) {
+  const invokedByPath: { [nodePath: string]: AspectApplication[] } = { };
+
+  recurse(root, []);
+
+  for (let i = 0; i <= 100; i++) {
+    const didAnythingToTree = recurse(root, []);
+
+    if (!didAnythingToTree) {
+      return;
+    }
+  }
+
+  throw new Error('We have detected a possible infinite loop while invoking Aspects. Please check your Aspects and verify there is no configuration that would cause infinite Aspect or Node creation.');
+
+  function recurse(construct: IConstruct, inheritedAspects: AspectApplication[]): boolean {
+    const node = construct.node;
+    const aspects = Aspects.of(construct);
+
+    let didSomething = false;
+
+    let localAspects = aspects.list;
+    const allAspectsHere = sortAspectsByPriority(inheritedAspects, localAspects);
+
+    for (const aspectApplication of allAspectsHere) {
+
+      let invoked = invokedByPath[node.path];
+      if (!invoked) {
+        invoked = invokedByPath[node.path] = [];
+      }
+
+      if (invoked.some(invokedApp => invokedApp.aspect === aspectApplication.aspect)) {
+        continue;
+      }
+
+      // If the last invoked Aspect has a higher priority than the current one, throw an error:
+      const lastInvokedAspect = invoked[invoked.length - 1];
+      if (lastInvokedAspect && lastInvokedAspect.priority > aspectApplication.priority) {
+        throw new Error(
+          `Cannot invoke Aspect ${aspectApplication.aspect.constructor.name} with priority ${aspectApplication.priority} after Aspect ${lastInvokedAspect.aspect.constructor.name} with priority ${lastInvokedAspect.priority} at ${node.path}.`,
```

Comment:
I think I told you to frame this in terms of what the user did to end up in this situation, and what they can do to fix it?

This error message is a bit like:

* You go "I want to withdraw $500 from this account"
* System goes "I couldn't enqueue a request."

This may be true, but what *would* have been a more helpful message would have been "your balance is $400 which is lower than $500".

Reply:
Fixed!